store/datasets_test.go

```diff
@@ -0,0 +1,344 @@
+							Properties: map[string]interface{}{
+								"label": "Overall index",
+							},
+						},
+					}, nil, nil
+				}
+				return nil, nil, io.EOF
+			},
+			CloseFunc: closeNoErr,
+		}
+		conn := &mock.ConnMock{
+			CloseFunc: closeNoErr,
+			QueryNeoFunc: func(query string, params map[string]interface{}) (golangNeo4jBoltDriver.Rows, error) {
+				return rows, nil
+			},
+		}
+		pool := &mock.DBPoolMock{
+			OpenPoolFunc: func() (golangNeo4jBoltDriver.Conn, error) {
+				return conn, nil
+			},
+		}
+		neo := NeoDataStore{
+			pool:          pool,
+			codeListLabel: "",
+		}
+
+		Convey("when I request the datasets from the store", func() {
+			datasets, err := neo.GetCodeDatasets(context.Background(), "my-code-list-id", "2017", "my-code")
+			Convey("then the datasets are returned correctly", func() {
+				So(err, ShouldBeNil)
+				So(datasets.NumberOfResults, ShouldEqual, 2)
+
+				item1 := datasets.Items[0]
+				So(item1.DimensionLabel, ShouldEqual, "Overall index")
+				So(item1.ID, ShouldEqual, "cpih01")
+				So(item1.Edition, ShouldEqual, "time-series")
+				So(item1.Version, ShouldEqual, 1)
+				So(item1.Links.CodeEdition.Href, ShouldEqual, "/code-lists/my-code-list-id/editions/2017")
+				So(item1.Links.DatasetDimension.Href, ShouldEqual, "/datasets/cpih01/editions/time-series/versions/1/dimensions/my-code-list-id")
+				So(item1.Links.DatasetVersion.Href, ShouldEqual, "/datasets/cpih01/editions/time-series/versions/1")
+
+				item2 := datasets.Items[1]
+				So(item2.DimensionLabel, ShouldEqual, "Overall index")
+				So(item2.ID, ShouldEqual, "mid-year-pop-est")
+				So(item2.Edition, ShouldEqual, "time-series")
+				So(item2.Version, ShouldEqual, 2)
+				So(item2.Links.CodeEdition.Href, ShouldEqual, "/code-lists/my-code-list-id/editions/2017")
+				So(item2.Links.DatasetDimension.Href, ShouldEqual, "/datasets/mid-year-pop-est/editions/time-series/versions/2/dimensions/my-code-list-id")
+				So(item2.Links.DatasetVersion.Href, ShouldEqual, "/datasets/mid-year-pop-est/editions/time-series/versions/2")
+			})
+		})
+	})
+
+	Convey("given a request for code datasets is made for an edition that does not exist", t, func() {
+		rows := &mock.RowsMock{
+			NextNeoFunc: func() ([]interface{}, map[string]interface{}, error) {
+				return nil, nil, io.EOF
+			},
+			CloseFunc: closeNoErr,
+		}
+		conn := &mock.ConnMock{
+			CloseFunc: closeNoErr,
+			QueryNeoFunc: func(query string, params map[string]interface{}) (golangNeo4jBoltDriver.Rows, error) {
+				return rows, nil
+			},
+		}
+		pool := &mock.DBPoolMock{
+			OpenPoolFunc: func() (golangNeo4jBoltDriver.Conn, error) {
+				return conn, nil
+			},
+		}
+		neo := NeoDataStore{
+			pool:          pool,
+			codeListLabel: "",
+		}
+		Convey("when the request is made", func() {
+			datasets, err := neo.GetCodeDatasets(context.Background(), "", "", "")
+			Convey("then an error is returned", func() {
+				So(err, ShouldNotBeNil)
+				So(err.Error(), ShouldEqual, "edition not found")
```

Comment:
Consider asserting the error type instead of the error string: `So(err, ShouldEqual, datastore.ErrEditionNotFound)`